pkg/patterns/declarative/reconciler.go

```diff
@@ -166,53 +178,63 @@ func (r *Reconciler) BuildDeploymentObjects(ctx context.Context, name types.Name
 	log := log.Log
 
 	// 1. Load the manifest
-	manifestStr, err := r.loadRawManifest(ctx, instance)
+	manifestFiles, err := r.loadRawManifest(ctx, instance)
 	if err != nil {
 		log.Error(err, "error loading raw manifest")
 		return nil, err
 	}
-
+	mainfestObjects := &manifest.Objects{}
 	// 2. Perform raw string operations
-	for _, t := range r.options.rawManifestOperations {
-		transformed, err := t(ctx, instance, manifestStr)
+	for filepath, manifestStr := range manifestFiles {
+		for _, t := range r.options.rawManifestOperations {
+			transformed, err := t(ctx, instance, manifestStr)
+			if err != nil {
+				log.Error(err, "error performing raw manifest operations")
+				return nil, err
+			}
+			manifestStr = transformed
+		}
+		// 3. Parse manifest into objects
+		objects, err := manifest.ParseObjects(ctx, manifestStr)
 		if err != nil {
-			log.Error(err, "error performing raw manifest operations")
+			log.Error(err, "error parsing manifest")
 			return nil, err
 		}
-		manifestStr = transformed
-	}
-
-	// 3. Parse manifest into objects
-	objects, err := manifest.ParseObjects(ctx, manifestStr)
-	if err != nil {
-		log.Error(err, "error parsing manifest")
-		return nil, err
-	}
 
-	// 4. Perform object transformations
-	transforms := r.options.objectTransformations
-	if r.options.labelMaker != nil {
-		transforms = append(transforms, AddLabels(r.options.labelMaker(ctx, instance)))
-	}
-	// TODO(jrjohnson): apply namespace here
-	for _, t := range transforms {
-		err := t(ctx, instance, objects)
-		if err != nil {
-			return nil, err
+		// 4. Perform object transformations
+		transforms := r.options.objectTransformations
+		if r.options.labelMaker != nil {
+			transforms = append(transforms, AddLabels(r.options.labelMaker(ctx, instance)))
 		}
+		// TODO(jrjohnson): apply namespace here
+		for _, t := range transforms {
+			err := t(ctx, instance, objects)
+			if err != nil {
+				return nil, err
+			}
+		}
+		// 5. Sort objects to work around dependent objects in the same manifest (eg: service-account, deployment)
```

Comment:
Can you move the Sort step outside of the new for loop (step 2)? That way we get a consistent manifest every time. Sorting within a single file may not accomplish that.